src/librustc/middle/expr_use_visitor.rs

```diff
@@ -608,9 +608,10 @@ impl<'a, 'gcx, 'tcx> ExprUseVisitor<'a, 'gcx, 'tcx> {
     fn walk_local(&mut self, local: &hir::Local) {
         match local.init {
             None => {
-                let delegate = &mut self.delegate;
-                local.pat.each_binding(|_, id, span, _| {
-                    delegate.decl_without_init(id, span);
+                local.pat.each_binding(|_, hir_id, span, _| {
+                    // FIXME: converting HirId → NodeId is said to be relatively expensive
+                    let node_id = self.mc.tcx.hir.definitions().find_node_for_hir_id(hir_id);
```

Comment:
You can use `self.mc.tcx.hir.hir_to_node_id()` which is "just" a hash table lookup. `find_node_for_hir_id()` should actually be removed.